cmd/run.go

```diff
@@ -115,19 +120,79 @@ func run(infile string) {
 
 	args = append([]string{"-m", outfile}, args...)
 	stdout, stderr, file := logging()
-	gjs := lib.Exec("gjs", args...)
-	gjs.Stdin = os.Stdin
-	gjs.Dir = filepath.Dir(infile)
+	if file != nil {
+		defer file.Close()
+	}
+
+	var gjs *exec.Cmd
+	gjsStart := func() (*exec.Cmd, error) {
+		if gjs != nil {
+			if err := gjs.Process.Kill(); err != nil {
+				log.Printf("Failed to kill process: %v", err)
+			}
+			_ = gjs.Wait()
+		}
+
+		gjs = lib.Exec("gjs", args...)
+		gjs.Stdin = os.Stdin
+		gjs.Dir = filepath.Dir(infile)
 
-	gjs.Stdout = stdout
-	gjs.Stderr = stderr
+		gjs.Stdout = stdout
+		gjs.Stderr = stderr
 
-	// TODO: watch and restart
-	if err := gjs.Run(); err != nil {
+		err := gjs.Start()
+
+		return gjs, err
+	}
+
+	if err := watchRun(gjsStart); err != nil {
 		lib.Err(err)
 	}
+}
 
-	if file != nil {
-		file.Close()
+func watchRun(gjsStart func() (*exec.Cmd, error)) error {
+	cmd, err := gjsStart()
+	if err != nil {
+		return err
 	}
+	if !watch {
+		return cmd.Wait()
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return err
+	}
+	defer watcher.Close()
+
+	go func() {
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return
+				}
+				log.Println("event:", event)
+				if event.Has(fsnotify.Write) {
```

Comment:
[`Remove` events](https://pkg.go.dev/github.com/fsnotify/fsnotify#Watcher) don't seem like they'll be caught here (rename and create ones will also fire a Write it looks like). That's probably fine but worth a comment IMO